src/electron/routing_service.ts

```diff
@@ -82,16 +82,22 @@ export class RoutingDaemon {
     return new Promise<void>((fulfill, reject) => {
       const newSocket = this.socket = createConnection(SERVICE_NAME, () => {
         newSocket.removeListener('error', initialErrorHandler);
+        const cleanup = () => {
+          newSocket.removeAllListeners();
+          this.fulfillDisconnect();
+        };
+        newSocket.once('close', cleanup);
+        newSocket.once('error', cleanup);
 
         newSocket.once('data', (data) => {
-          const message: RoutingServiceResponse = JSON.parse(data.toString());
-          if (message.action !== RoutingServiceAction.CONFIGURE_ROUTING ||
+          const message = this.parseRoutingServiceResponse(data);
+          if (!message || message.action !== RoutingServiceAction.CONFIGURE_ROUTING ||
               message.statusCode !== RoutingServiceStatusCode.SUCCESS) {
             // NOTE: This will rarely occur because the connectivity tests
             //       performed when the user clicks "CONNECT" should detect when
             //       the system is offline and that, currently, is pretty much
             //       the only time the routing service will fail.
-            reject(new Error(message.errorMessage));
+            reject(new Error(!!message ? message.errorMessage : 'empty routing service response'));
```

Comment:
It might be nicer to just have !!message in a separate branch

Reply:
Tried a few refactors, but they end up making the code less readable or duplicated IMO.